### PR TITLE
[KOGITO-1998] - Deprecate KogitoApp CR

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Jira issue : https://issues.redhat.com/browse/KOGITO-1998

Depends on https://github.com/kiegroup/kogito-cloud-operator/pull/479

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]
LGTM! Thanks for this, @vaibhavjainwiz 
Where you able to test some of them in an actual cluster?

**Comment by vaibhavjainwiz** [comment]
> LGTM! Thanks for this, @vaibhavjainwiz
> Where you able to test some of them in an actual cluster?

yes.. I had tested `process-quarkus-example` & `process-springboot-example` in actual OS cluster

**Review by danielezonca — APPROVED** [review]
LGTM

File: process-quarkus-example/operator/process-quarkus-example.yaml [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+kind: KogitoRuntime
+metadata:
+  name: process-quarkus-example
+# uncomment to enable persistence
+#spec:
+#  infinispan:
+#    useKogitoInfra: true
+  #envs:
+  #  - name: MAVEN_ARGS_APPEND
+  #    value: "-Ppersistence"1
```
Comment (sutaakar) [inline process-quarkus-example/operator/process-quarkus-example.yaml]:
```suggestion
  #    value: "-Ppersistence"
```

**Review by sutaakar — APPROVED** [review]

File: process-kafka-quickstart-quarkus/operator/process-kafka-quickstart-quarkus.yaml [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+  # set your maven nexus repository to speed up the build time
+  #mavenMirrorURL:
+---
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoRuntime
+metadata:
+  name: process-kafka-quickstart-quarkus
+spec:
+  kafka:
+    UseKogitoInfra: true
```
Comment (radtriste) [inline process-kafka-quickstart-quarkus/operator/process-kafka-quickstart-quarkus.yaml]:
```suggestion
    useKogitoInfra: true
```

File: process-kafka-quickstart-springboot/operator/process-kafka-quickstart-springboot.yaml [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
+  #mavenMirrorURL:
+---
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoRuntime
+metadata:
+  name: process-kafka-quickstart-springboot
+spec:
+  runtime: springboot
+  kafka:
+    UseKogitoInfra: true
```
Comment (radtriste) [inline process-kafka-quickstart-springboot/operator/process-kafka-quickstart-springboot.yaml]:
```suggestion
    useKogitoInfra: true
```

File: kogito-travel-agency/extended/operator/travels.yaml [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+---
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoRuntime
+metadata:
+  name: travels
+spec:
+  infinispan:
+    useKogitoInfra: true
+  kafka:
+    UseKogitoInfra: true
```
Comment (radtriste) [inline kogito-travel-agency/extended/operator/travels.yaml]:
```suggestion
    useKogitoInfra: true
```

File: kogito-travel-agency/extended/operator/visas.yaml [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+---
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoRuntime
+metadata:
+  name: visas
+spec:
+  infinispan:
+    useKogitoInfra: true
+  kafka:
+    UseKogitoInfra: true
```
Comment (radtriste) [inline kogito-travel-agency/extended/operator/visas.yaml]:
```suggestion
    useKogitoInfra: true
```

File: serverless-workflow-events-quarkus/operator/serverless-workflow-events-quarkus.yaml [hunk 6]
```diff
@@ -1,16 +1,23 @@
...
+  # set your maven nexus repository to speed up the build time
+  #mavenMirrorURL:
+---
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoRuntime
+metadata:
+  name: serverless-workflow-events-quarkus
+spec:
+  kafka:
+    UseKogitoInfra: true
```
Comment (radtriste) [inline serverless-workflow-events-quarkus/operator/serverless-workflow-events-quarkus.yaml]:
```suggestion
    useKogitoInfra: true
```

**Review by radtriste — CHANGES_REQUESTED** [review]

File: process-quarkus-example/operator/process-quarkus-example.yaml (hunk 1 above)
Comment (vaibhavjainwiz) [reply process-quarkus-example/operator/process-quarkus-example.yaml]: done

File: process-kafka-quickstart-quarkus/operator/process-kafka-quickstart-quarkus.yaml (hunk 2 above)
Comment (vaibhavjainwiz) [reply process-kafka-quickstart-quarkus/operator/process-kafka-quickstart-quarkus.yaml]: done

File: kogito-travel-agency/extended/operator/travels.yaml (hunk 4 above)
Comment (vaibhavjainwiz) [reply kogito-travel-agency/extended/operator/travels.yaml]: done

File: serverless-workflow-events-quarkus/operator/serverless-workflow-events-quarkus.yaml (hunk 6 above)
Comment (vaibhavjainwiz) [reply serverless-workflow-events-quarkus/operator/serverless-workflow-events-quarkus.yaml]: done

File: kogito-travel-agency/extended/operator/visas.yaml (hunk 5 above)
Comment (vaibhavjainwiz) [reply kogito-travel-agency/extended/operator/visas.yaml]: done

File: process-kafka-quickstart-springboot/operator/process-kafka-quickstart-springboot.yaml (hunk 3 above)
Comment (vaibhavjainwiz) [reply process-kafka-quickstart-springboot/operator/process-kafka-quickstart-springboot.yaml]: done

**Review by radtriste — APPROVED** [review]

**Comment by ricardozanini** [comment]
@spolti please give us your blessing so I could merge this one

**Comment by spolti** [comment]
sry, missed this one.

**Review by spolti — APPROVED** [review]
sounds good, thanks @vaibhavjainwiz

**Comment by ricardozanini** [comment]
Once https://github.com/kiegroup/kogito-cloud-operator/pull/479 is merged, we can merge this.

**Pull request merged**